pytest.ini

```diff
@@ -13,12 +13,6 @@ addopts =
     --showlocals
     --doctest-modules
     --junitxml=.test-results/pytest/results.xml
-
-    # `pytest-cov`:
-    --cov=ansiblelint
-    --cov-report term-missing:skip-covered
-    --cov-report xml:.test-results/pytest/cov.xml
-    --no-cov-on-fail
```

Comment:
Have you considered keeping these here with `--no-cov` and just re-enabling in `tox.ini` (by setting `--cov`)? This would allow keeping more config settings consolidated.

Reply:
I did but it would be a mistake. It will fail to run if user does not have coverage plugin installed. Sometimes I do not have it and TBH, I do not see a need to add it as a runtime requirement for cases that do not need it. That is another reason why adding any plugin specific options in defaults is dangerous, as it prevents running pytest without them: aka xdist options, some kind of problem.

Our tests should be to run in absence of all/most pytest plugins, as these are supposed to add extra features, but still nicely degrade in their absence. xdist and cov fit this category.

Reply:
Well, you know that I'm against supporting this many ways to run the tests, this does not sound convincing.